data-prepper-plugins/s3-source/src/integrationTest/java/org/opensearch/dataprepper/plugins/source/SqsWorkerIT.java

```diff
@@ -127,6 +147,145 @@ void processSqsMessages_should_return_at_least_one_message(final int numberOfObj
+    @ValueSource(ints = {1})
+    void processSqsMessages_should_return_at_least_one_message_with_acks_with_callback_invoked_after_processS3Object_finishes(final int numberOfObjectsToWrite) throws IOException {
+        writeToS3(numberOfObjectsToWrite);
+
+        when(s3SourceConfig.getAcknowledgements()).thenReturn(true);
+        final Counter receivedCounter = mock(Counter.class);
+        final Counter deletedCounter = mock(Counter.class);
+        final Counter ackCallbackCounter = mock(Counter.class);
+        when(pluginMetrics.counter(SqsWorker.SQS_MESSAGES_RECEIVED_METRIC_NAME)).thenReturn(receivedCounter);
+        when(pluginMetrics.counter(SqsWorker.SQS_MESSAGES_DELETED_METRIC_NAME)).thenReturn(deletedCounter);
+        when(pluginMetrics.counter(SqsWorker.ACKNOWLEDGEMENT_SET_CALLACK_METRIC_NAME)).thenReturn(ackCallbackCounter);
+        lenient().doAnswer((val) -> {
+            receivedCount += (double)val.getArgument(0);
+            return null;
+        }).when(receivedCounter).increment(any(Double.class));
+        lenient().doAnswer((val) -> {
+            if (val.getArgument(0) != null) {
+                deletedCount += (double)val.getArgument(0);
+            }
+            return null;
+        }).when(deletedCounter).increment(any(Double.class));
+        lenient().doAnswer((val) -> {
+            ackCallbackCount += 1;
+            return null;
+        }).when(ackCallbackCounter).increment();
+
+        doAnswer((val) -> {
+            AcknowledgementSet ackSet = val.getArgument(1);
+            S3ObjectReference s3ObjectReference = val.getArgument(0);
+            assertThat(s3ObjectReference.getBucketName(), equalTo(bucket));
+            assertThat(s3ObjectReference.getKey(), startsWith("s3 source/sqs/"));
+            event = (Event)JacksonEvent.fromMessage(val.getArgument(0).toString());
+            ackSet.add(event);
+            return null;
+        }).when(s3Service).addS3Object(any(S3ObjectReference.class), any(AcknowledgementSet.class));
+        ExecutorService executor = Executors.newFixedThreadPool(2);
+        acknowledgementSetManager = new  DefaultAcknowledgementSetManager(executor);
+        final SqsWorker objectUnderTest = createObjectUnderTest();
+        Thread sinkThread = new Thread(() -> {
+            try {
+                synchronized(this) {
+                    while (!ready.get()) {
+                        Thread.sleep(100);
+                        this.wait();
+                    }
+                    if (event.getEventHandle() != null) {
+                        event.getEventHandle().release(true);
+                    }
+                }
+            } catch (Exception e){}
+        });
+        sinkThread.start();
+        final int sqsMessagesProcessed = objectUnderTest.processSqsMessages();
+        synchronized(this) {
+            ready.set(true);
+            this.notify();
+        }
+        try {
+            Thread.sleep(10000);
+        } catch (Exception e){}
```

Comment:
Are you expecting the thread to be interrupted? If you are not expecting this, then you can just let an interruption fail the test.

```
Thread.sleep(10000);
```

and update the method to `throws Exception`.